network-libp2p/src/discovery/handler.rs

```diff
@@ -244,6 +251,23 @@ impl Handler {
     }
 }
 
+/// Extract the `/ip4/`,`/ip6/`, `/dns4/` or `/dns6/` protocol part from a `Multiaddr`
+pub(crate) fn protocol_part(addr: &Multiaddr) -> Option<Multiaddr> {
```

Comment:
Mmm, maybe `outer_protocol_address`? Because there are other `protocols` on top of this outer protocols.

Reply:
We also need to be careful here when adding circuits, since we would be getting the relay address

Reply:
Correct, a valid CR address is `/ip4/198.51.100.0/tcp/55555/p2p/QmRelay/p2p-circuit/p2p/QmAlice`.
Everything prior to the `/p2p-circuit/` is the address of the relay which we don't want to ban of course. But that is a concern for a later PR.